src/api/src/services/LogFile/index.ts

```diff
@@ -0,0 +1,31 @@
+import { Service } from 'typedi';
+import { LoggerService } from '../../logger';
+import LogEntry from '../../models/LogEntry';
+
+export interface ILogFileArgs {
+  numberOfLines: number;
+}
+
+export interface ILogFile {
+  loadLogFile(args: ILogFileArgs): Promise<LogEntry[] | null>;
+}
+
+@Service()
+export default class LogFileService implements ILogFile {
+  constructor(private logger: LoggerService) {}
+
+  async loadLogFile(args: ILogFileArgs) {
+    const contentPromise = new Promise<LogEntry[] | null>((resolve) => {
+      this.logger.log('Requested log content');
+      this.logger.query(
```

Comment:
Once logger.query is converted to async method this loadLogFile() method will look much cleaner